unit/solvers/smt2_incremental/smt_index.cpp

```diff
@@ -0,0 +1,69 @@
+// Author: Diffblue Ltd.
+
+#include <util/optional.h>
+
+#include <solvers/smt2_incremental/smt_index.h>
+#include <testing-utils/use_catch.h>
+
+TEST_CASE("Test smt_indext.pretty is accessible.", "[core][smt2_incremental]")
+{
+  const smt_indext index = smt_numeral_indext{42};
+  REQUIRE(index.pretty(0, 0) == "smt_numeral_index\n  * value: 42");
```

Comment:
This is based on the assumption that `.pretty()` is always going to return text of a specific format.

I'm not sure if this assumption is always supposed to hold.

If the aim of the test is to make sure that `pretty()` is accessible, why not just make sure that it's not the empty string? And if you need to make sure that the value is there, make a substring search on that?